utils/mcp_client.py

```
@@ -0,0 +1,531 @@
+        or parsed.is_link_local
+        or parsed.is_multicast
+        or parsed.is_reserved
+        or parsed.is_unspecified
+    )
+
+
+def validate_mcp_http_url(url: str) -> None:
+    """Block obvious SSRF targets unless explicitly allowed."""
+    parsed = urlparse(url)
+    if parsed.scheme not in {"http", "https"} or not parsed.hostname:
+        raise McpClientError(f"Invalid MCP HTTP URL: {url}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)